aptos-move/aptos-vm/src/move_vm_ext/write_op_converter.rs

```diff
@@ -186,8 +185,15 @@ impl<'r> WriteOpConverter<'r> {
                 Op::New(bytes)
             };
 
-            // TODO(loader_v2): Query state value metadata from module storage.
-            let write_op = self.convert_module(&state_key, op, false)?;
+            let state_value_metadata = module_storage.fetch_state_value_metadata(addr, name)?;
```

Comment:
Let's test this beautiful method, now that we purely depend on module storage interfaces. We can make sure new / modify cases work with the correct metadata.

Reply:
done